runtime/src/iree/base/internal/cpu.h

```diff
@@ -10,13 +10,56 @@
 #include <stddef.h>
 
 #include "iree/base/api.h"
+#include "iree/schemas/cpu_data.h"
 
 #ifdef __cplusplus
 extern "C" {
 #endif  // __cplusplus
 
 //===----------------------------------------------------------------------===//
-// iree_cpu_*
+// Processor data query
+//===----------------------------------------------------------------------===//
+
+// Initializes cached CPU data using |temp_allocator| for any temporary
+// allocations required during initialization.
+void iree_cpu_initialize(iree_allocator_t temp_allocator);
+
+// Initializes cached CPU data with the given fields.
+// Extraneous fields will be ignored and unspecified fields will be set to zero.
+void iree_cpu_initialize_with_data(iree_host_size_t field_count,
+                                   const uint64_t* fields);
+
+// Returns all fields up to IREE_CPU_DATA_FIELD_COUNT.
+// Data will be zeroed until initialized with iree_cpu_initialize.
+// See iree/schemas/cpu_data.h for interpretation.
+const uint64_t* iree_cpu_data_fields(void);
```

Comment:
I don't understand what this ability to export raw cached data is needed for. If we only had opaque `iree_cpu_initialize` and symbolic-constant-based bit lookup, wouldn't that be enough and avoid the need to allocate a somewhat larger, arbitrary-sized `iree_cpu_data_cache_` ?

Reply:
You cannot call these functions from the ukernel library or generated code and the fields need to be passed in to such code in order to be used. This allows access to the fields so that they can be passed in.